R/feature.align.R

```diff
@@ -124,29 +121,41 @@ create_rows <- function(features,
 #'  percentage of the m/z value. This value, multiplied by the m/z value, becomes the cutoff level.
 #' @param rt_tol_relative The retention time tolerance level for peak alignment. The default is NA, which
 #'  allows the program to search for the tolerance level based on the data.
+#' @param cluster The number of CPU cores to be used
 #' @return A tibble with three tables containing aligned metadata, intensities an RTs.
 #'
 #' @export
 create_aligned_feature_table <- function(features_table,
                                          min_occurrence,
                                          sample_names,
                                          rt_tol_relative,
-                                         mz_tol_relative) {
-    
+                                         mz_tol_relative,
+                                         cluster = 4) {
+    if (!is(cluster, "cluster")) {
+        cluster <- parallel::makeCluster(cluster)
+        on.exit(parallel::stopCluster(cluster))
+    }
+
+    # NOTE: side effect (doParallel has no functionality to clean up)
```

Comment:
What does it mean `no functionality to clean up`?

Reply:
That it there is no functionality to end the parallel system or cluster created afterwards